assets/test/spec/services/apiServiceSpec.js

```diff
@@ -0,0 +1,204 @@
+      [{kind:"Pod",      apiVersion:"v1"}, {g:'',v:'v1',r:'pods'}],
+
+      // extensions
+      [{kind:"Job",apiVersion:"extensions/v1beta1"}, {g:'extensions',v:'v1beta1',r:'jobs'}],
+      [{kind:"Foo",apiVersion:"unknown/v1beta6"},    {g:'unknown',   v:'v1beta6',r:'foos'}],
+    ];
+    angular.forEach(tc, function(item) {
+      it('should result in ' + JSON.stringify(item[1]) + ' when called with ' + JSON.stringify(item[0]), function() {
+        var expectedRGV = item[1];
+        
+        // Call once and compare the components
+        var actualRGV = APIService.objectToResourceGroupVersion(item[0]);
+        if (expectedRGV) {
+          expect(actualRGV.resource).toEqual(expectedRGV.r);
+          expect(actualRGV.group   ).toEqual(expectedRGV.g);
+          expect(actualRGV.version ).toEqual(expectedRGV.v);
+        } else {
+          expect(actualRGV).toEqual(expectedRGV);
+        }
+      });
+    });
+  });  
+  
+  
+  
+  describe("#kindToResource", function(){
+    var tc = [
+      // invalid cases
+      [null,              ""],
+      ["",                ""],
+      
+      // pluralization
+      ["foo",             "foos"],
+      // pluralization with s
+      ["foos",            "fooses"],
+      // pluralization with y
+      ["Policy",          "policies"],
+      // special cases
+      ["Endpoints",                  "endpoints"],
+      ["SecurityContextConstraints", "securitycontextconstraints"],
+    ];
+    angular.forEach(tc, function(item) {
+      it('should result in ' + JSON.stringify(item[1]) + ' when called with ' + JSON.stringify(item[0]), function() {
+        expect(APIService.kindToResource(item[0])).toEqual(item[1]);
+      });
+    });
+  });  
+
+  describe("#deriveTargetResource", function(){
+    var tc = [
+      // invalid cases
+      [null,null,              undefined],
+      ["","",                  undefined],
+      [{},{},                  undefined],
+      
+      // simple resource, matching object overrides group/version
+      ['pods', {kind:"Pod",apiVersion:"v1"},                 {r:'pods',g:'',          v:'v1'     }],
+      ['pods', {kind:"Pod",apiVersion:"v2"},                 {r:'pods',g:'',          v:'v2'     }],
+      ['jobs', {kind:"Job",apiVersion:"extensions/v1beta1"}, {r:'jobs',g:'extensions',v:'v1beta1'}],
+      ['jobs', {kind:"Job",apiVersion:"extensions/v1beta2"}, {r:'jobs',g:'extensions',v:'v1beta2'}],
+
+      // simple resource, non-matching object leaves group/version alone
+      ['pods', {kind:"Foo",apiVersion:"v1"},                 {r:'pods',g:'',v:'v1'}],
+      ['pods', {kind:"Foo",apiVersion:"v2"},                 {r:'pods',g:'',v:'v1'}],
+      ['jobs', {kind:"Foo",apiVersion:"extensions/v1beta1"}, {r:'jobs',g:'',v:'v1'}],
+      ['jobs', {kind:"Foo",apiVersion:"extensions/v1beta2"}, {r:'jobs',g:'',v:'v1'}],
+      // actual use:
+      ['deploymentconfigs/scale', {kind:"Scale",apiVersion:"extensions/v1beta1"}, {r:'deploymentconfigs/scale',g:'',v:'v1'}],
```

Comment:
It's working. Look correct?

```js
      var scale = {
        apiVersion: "extensions/v1beta1",
        kind: "Scale",
        metadata: {
          name: dc.metadata.name,
          namespace: dc.metadata.namespace
        },
        spec: {
          replicas: replicas
        }
      };
      return DataService.update("deploymentconfigs/scale", dc.metadata.name, scale, {
        namespace: dc.metadata.namespace
      });
```

<img width="658" alt="openshift_web_console" src="https://cloud.githubusercontent.com/assets/1167259/12822683/2f07080a-cb37-11e5-8f89-640db34a4e39.png">